### ISSUE TITLE
metagpt.ext.aflow.scripts.optimizer:optimize:93 - Error occurred: object list can't be used in 'await' expression. Retrying... (Attempt 1/1)

### ISSUE THREAD

**Issue by CALMCRAZY** [geekan/MetaGPT#1597]
**Bug description**
metagpt.ext.aflow.scripts.optimizer:optimize:93 - Error occurred: object list can't be used in 'await' expression. Retrying... (Attempt 1/1)

**Bug solved method**
not solved

An error occurred when running AFLOW with this command:
```
python -m examples.aflow.optimize --dataset custom --max_rounds 1 --validation_rounds 3 --sample 2
```

**Environment information**
- LLM type and model name:OpenAI grok-beta
- System version:Windows 11
- Python version:conda python 3.11.1
- MetaGPT version or branch:2024.11 main

- packages version:unknown
- installation method: pip install -e .

**Screenshots or logs**
2024-11-13 17:12:16.364 | INFO     | metagpt.const:get_metagpt_package_root:21 - Package root set to B:\software\MetaGPT
2024-11-13 17:12:20.381 | INFO     | metagpt.ext.aflow.scripts.optimizer:optimize:93 - Error occurred: object list can't be used in 'await' expression. Retrying... (Attempt 1/1)
2024-11-13 17:12:20.382 | INFO     | metagpt.ext.aflow.scripts.optimizer:optimize:95 - Max retries reached. Moving to next round.
2024-11-13 17:12:25.384 | INFO     | metagpt.ext.aflow.scripts.optimizer:optimize:106 - Score for round 2: None

**Comment by didiforgithub** [comment]
Hi, thanks for your usage.

For this question, maybe you need to first determine that the custom dataset is set in DatasetType.

Subsequently, you can check your eval function to ensure that graph_evaluate returns the correct score, avg_cost, and total_cost. If the problem still cannot be resolved, you can provide more information.

**Comment by CALMCRAZY** [comment]
> ost 和 t

I checked what you mentioned:
The DatasetType configuration is correct.
The return value of the evaluation function meets expectations.

It seems that the run_evaluation method of BaseBenchmark erroneously awaited the synchronized load_data method. Load_data is a synchronization method that uses pandas to read files. It may not be necessary or appropriate to use await. When I removed 'await' and the bug disappeared, I had to observe for a while longer.
